src/replication_core.rs

```diff
@@ -20,57 +20,88 @@ impl Plugin for ReplicationCorePlugin {
+}
+
+impl Default for NetworkChannels {
+    fn default() -> NetworkChannels {
+        NetworkChannels {
+            server: Vec::with_capacity(u8::MAX as usize),
+            client: Vec::with_capacity(u8::MAX as usize),
+        }
+    }
 }
 
 impl NetworkChannels {
     pub fn server_channels(&self) -> Vec<ChannelConfig> {
-        channel_configs(self.server)
+        channel_configs(&self.server)
     }
 
     pub fn client_channels(&self) -> Vec<ChannelConfig> {
-        channel_configs(self.client)
+        channel_configs(&self.client)
     }
 
-    pub(super) fn create_client_channel(&mut self) -> u8 {
-        self.client += 1;
-        self.client + REPLICATION_CHANNEL_ID
+    pub(super) fn create_client_channel(&mut self, send_policy: SendPolicy) -> u8 {
+        if self.client.len() == (REPLICATION_CHANNEL_ID + u8::MAX).into() {
+            panic!("Max client channels exceeded.");
+        }
+        self.client.push(send_policy);
+        self.client.len() as u8 + REPLICATION_CHANNEL_ID
     }
 
-    pub(super) fn create_server_channel(&mut self) -> u8 {
-        self.server += 1;
-        self.server + REPLICATION_CHANNEL_ID
+    pub(super) fn create_server_channel(&mut self, send_policy: SendPolicy) -> u8 {
+        if self.server.len() == (REPLICATION_CHANNEL_ID + u8::MAX).into() {
+            panic!("Max server channels exceeded.");
+        }
+        self.server.push(send_policy);
+        self.server.len() as u8 + REPLICATION_CHANNEL_ID
     }
 }
 
-fn channel_configs(events_count: u8) -> Vec<ChannelConfig> {
-    let mut channel_configs = Vec::with_capacity((events_count + 1).into());
+fn channel_configs(channels: &Vec<SendPolicy>) -> Vec<ChannelConfig> {
+    let mut channel_configs = Vec::with_capacity(channels.len() + 1);
     // TODO: Make it configurable.
     // Values from `DefaultChannel::config()`.
     channel_configs.push(ChannelConfig {
         channel_id: REPLICATION_CHANNEL_ID,
         max_memory_usage_bytes: 5 * 1024 * 1024,
         send_type: SendType::Unreliable,
     });
-    for channel_id in 1..=events_count {
+    for (idx, send_policy) in channels.iter().enumerate() {
         channel_configs.push(ChannelConfig {
-            channel_id: REPLICATION_CHANNEL_ID + channel_id,
+            channel_id: REPLICATION_CHANNEL_ID + 1 + idx as u8,
             max_memory_usage_bytes: 5 * 1024 * 1024,
-            send_type: SendType::ReliableOrdered {
-                resend_time: Duration::from_millis(300),
-            },
+            send_type: send_policy_to_type(*send_policy, Duration::from_millis(300)),
         });
     }
     channel_configs
 }
 
+fn send_policy_to_type(send_policy: SendPolicy, resend_time: Duration) -> SendType {
+    match send_policy {
+        SendPolicy::Unreliable => SendType::Unreliable,
+        SendPolicy::Unordered => SendType::ReliableUnordered { resend_time },
+        SendPolicy::Ordered => SendType::ReliableOrdered { resend_time },
+    }
+}
```

Comment:
We currently always use `Duration::from_millis(300)`. How about to implement `From` with this value hardcoded?

And with change we could accept `impl Into<SendType>` in our event functions. This way user will be able to pass our simple `SendPolicy` enum for default value and at the same time pass custom-configured `SendType`.
But I think we should give it a little more descriptive name. How about `EventPolicy`? Also short and won't be confusing with `SendType`.

Reply:
Okay!